### ISSUE TITLE
Similarity with react-autobind?

### ISSUE THREAD

**Issue by markerikson** [zackargyle/react-bind-decorator#4]
Looks pretty similar to https://github.com/cassiozen/React-autobind .  Any differences?

**Comment by zackargyle** [comment]
Looks like that is 1) not a decorator, and 2) bound at the instance level instead of the prototype level. So my guess is that it is much slower (similar in performance to the ES6ConstructorBindVersion test case).

**Comment by zackargyle** [comment]
Looks like I was right. It is actually the slowest implementation by far. These are the results for the basic use case (binding everything).

```
export class ReactAutobindVersion extends Component {
    constructor() {
        super();
        reactautobind(this);
    }
    fn1() {}
    fn2() {}
    fn3() {}
    render() {
        return <div onClick={this.fn1}>Hello world</div>;
    }
}
```

<img width="930" alt="screen shot 2016-04-02 at 6 33 30 pm" src="https://cloud.githubusercontent.com/assets/3934294/14230164/7d159988-f901-11e5-9932-8801ffefd9e1.png">

**Comment by markerikson** [comment]
Two questions, then:

1) How much practical difference is there going to be speed and performance-wise between these approaches in a real app?
2) Is there a way to use react-bind-decorator as a plain function if a project isn't set up to use decorators?

**Comment by markerikson** [comment]
Skimming the code... would I use it as:

``` js
import bind from "react-bind-decorator";
class MyComponent extends Component{};

bind()(MyComponent);
```

**Comment by zackargyle** [comment]
Yup! Might actually be nice to return `target` from the internal function so that you can do:
`export default bind()(MyComponent);`
If you want to contribute that would be an easy PR!
